### PR TITLE
Must do changes

### PR DESCRIPTION
**Pull request by xieanne**
1. Added animation for login (already shown) [Currently if we log off and go back to the login page, the login stuff shifted down due to the animation don't really know why, still investigating]

2. Changed patient form UI
image
image

3. Added gender (but did not add to database yet @ianyechen prob need to add that), currently only printing out the gender

4. Changed oncologist form UI, also added password match (but is there anyway when we entering the password we change it to like the dot dot dot thingy without actually showing the password???)
image

### CODE REVIEW

**Comment by xieanne** [comment]
Actually prob need help with on how to reset the animation whenever we log off and go back to the log in page

**Comment by xieanne** [comment]
UPDATE: fixed animation problem thanks to angus
Added password echo for the oncolgosit form
image

**Comment by xieanne** [comment]
alright, that animation problem happens again when we enter smth wrong
image

**Review by liu-allan — CHANGES_REQUESTED** [review]
Remove lines 563 - 568 and replace with this to display the missing input message before the consent box message:

```
        except Exception as er:
            er = str(er)
            if er != "Patient must provide consent to store data":
                msg = "Input fields must not be empty"
            else:
                msg = "Patient must provide consent to store data"
```

File: widget_pages/login.py [hunk 1]
```diff
@@ -232,6 +260,29 @@ def loginPushed(self):
                 self.parent().parent().is_admin_user = False
             self.showPatientListWindow()
 
+    def loginPageAnimation(self):
```
Comment (liu-allan) [inline widget_pages/login.py:263]: The form moves when the user inputs an incorrect password
https://user-images.githubusercontent.com/44624435/230967971-831a1c48-fa6f-4711-8113-829c9f5a7488.mp4

**Comment by ianyechen** [comment]
> Changed oncologist form UI, also added password match (but is there anyway when we entering the password we change it to like the dot dot dot thingy without actually showing the password???)

@liu-allan did something similar for the login page, maybe you can take a look at [this](https://github.com/liu-allan/Leukemia-Treatment-Application/blob/main/widget_pages/login.py#L165)?

File: widget_pages/patient_form.py [hunk 2]
```diff
@@ -67,57 +65,186 @@ class PatientFormWindow(QWidget):
...
+        self.birthdayEdit = QDateEdit()
+        self.birthdayEdit.setContentsMargins(0, 0, 0, 0)
+        self.birthdayEdit.setFont(QFont("Avenir", 15))
+        self.birthdayLayout.addWidget(self.birthdayEdit)
+        self.patientFormLayout.addLayout(self.birthdayLayout)
+
+        self.bodyLayout = QGridLayout()
+        self.bodyLayout.setContentsMargins(0, 0, 0, 0)
+
+        self.weightLabel = Label("Weight(kg)")
```
Comment (ianyechen) [inline widget_pages/patient_form.py:184]: Can we have a space in between weight and (kg) please, same as all the other labels that have this issue.

Comment (liu-allan) [reply widget_pages/patient_form.py]: done

**Pull request closed without merging**